src/components/TextField/TextField.tsx

```diff
@@ -394,6 +413,15 @@ class TextField extends React.PureComponent<CombinedProps, State> {
     onChange(String(newValue.toFixed(decimalPlaces)), this.state.id);
   };
 
+  private handleClearButtonPress = () => {
+    const {id} = this.state;
+    const {onChange} = this.props;
```

Comment:
This is mostly stylistic, but questionably more performant if destructured together, so you limit the number of times you reference `this`.

```tsx
const {state: {id}, props: {onChange}} = this;
```